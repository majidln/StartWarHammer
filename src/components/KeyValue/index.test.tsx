import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {render, waitFor} from '@testing-library/react-native';
import {KeyValue} from './index';

it('KeyValue Snapshot renders correctly', () => {
  const tree = renderer
    .create(<KeyValue title="name" value="Majid" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('KeyValue render in list correctly', async () => {
  const {findByTestId} = render(<KeyValue title="name" value="Majid" />);

  await waitFor(() => findByTestId('keyValueId'));
});
