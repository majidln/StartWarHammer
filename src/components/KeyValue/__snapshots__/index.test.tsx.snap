// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyValue Snapshot renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "center",
      "marginVertical": 4,
      "width": "100%",
    }
  }
  testID="keyValueId"
>
  <Text
    style={
      Object {
        "color": "#2c3e50",
        "fontSize": 18,
        "paddingRight": 4,
      }
    }
  >
    name
  </Text>
  <Text
    style={
      Object {
        "color": "#000",
        "fontSize": 18,
        "fontWeight": "bold",
      }
    }
  >
    Majid
  </Text>
</View>
`;
